recipe/shelf.py

```diff
@@ -310,6 +311,57 @@ def ingredient_from_dict(ingr_dict, table=''):
     return IngredientClass(*args, **ingr_dict)
 
 
+def parse_validated_condition(cnd, table=''):
+    """ Converts a validated field to sqlalchemy condition """
+    field = parse_validated_field(cnd['field'], table=table)
+    return cnd['_condition'](field)
+
+
+def parse_validated_field(fld, table=''):
+    """ Converts a validated field to sqlalchemy """
+    tablename = table.__name__
+    locals()[tablename] = table
```

Comment:
I've seen this `locals()[tablename]` thing in other code recently, what's it all about?